specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlSyncTask.json

```diff
@@ -0,0 +1,483 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Database Migration Service Resource Provider",
+    "version": "2022-11-30-preview"
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "MigrateMySqlAzureDbForMySqlSyncTaskInput": {
+      "type": "object",
+      "description": "Input for the task that migrates MySQL databases to Azure Database for MySQL for online migrations",
+      "properties": {
+        "sourceConnectionInfo": {
+          "description": "Connection information for source MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "targetConnectionInfo": {
+          "description": "Connection information for target Azure Database for MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "selectedDatabases": {
+          "type": "array",
+          "description": "Databases to migrate",
+          "items": {
+            "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlSyncDatabaseInput"
+          }
+        },
+        "makeSourceServerReadOnly": {
+          "type": "boolean",
+          "description": "Setting to set the source server read only",
+          "default": false
+        },
```

Comment:
We should not have makeSourceServerReadOnly for online migration. As then, there is no point of doing online.

Reply:
Also, same comment here that why do we have makesourceServerReadOnly as a separate property whereas enableConsistentBackup as optional?

Reply:
Both makeSourceServerReadOnly and enableConsistentBackup are parts of the current model. But since they have correct default values (false and true, correspondingly, in the backend code), I will remove them from the API spec.

As for the inconsistency of these attributes' location, this is how they are currently implemented and already used in prod.